### PR TITLE
[BOUNTY] Beecoin shop

### LINKED ISSUE #627: Code Bounty: Beecoin Shop
## Wanted: A place to spend my Beecoin collection.
Implement a functioning Beecoin shop which can be used to spend Beecoins on additional character slots and cosmetic items/features to be used from the character creator. Not expecting any specific cosmetics added, just the framework to put items in for future development and a few examples for testing. As such the easier it is to code new stuff into the shop the better.
### Required Feature Checklist
All these features **must** be included to accept the bounty.
- [x] A beecoin shop menu accessible from the OOC tab or character preferences menu
- [x] Option to access and equip cosmetic items from the character creator
- [x] Features/support for job-restricted cosmetics
- [x] Additional character slots, that are locked until purchased individually from the beecoin shop.
### Optional Feature Checklist
These features are not required, but I'll tip extra if they're done well.
- [x] Support for equipping cosmetics based on body region (Hat slot, jumpsuit slot, ect)
- [ ] Preview cosmetic options from the character creator before purchasing
- [x] Make the shop menu look nice / easy to navigate for players
- [x] Make the shop code easy to maintain / further develop
- [ ] Adding items you think are cool
Items currently fulfilled by #645 , bounty will be closed once the PR is accepted.

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

Implements a beecoin shop for cosmetic items, based on my unmerged loadouts port for FTL13. This PR is intentionally barebones in terms of items so this can be tested on the live server without draining everybody's beecoins accidentally and while prices are still potentially changing. A follow-up PR will shortly follow with the items I ripped out, as well as several more.

TODO:
- [x] Figure out decent prices for everything.

Closes #627 

## Changelog
:cl:
add: You can now spend your hard-earned beecoins in the new Loadout shop in the preferences window. It'll be pretty barebones until the feature has been well-tested.
add: WARNING: The loadout shop is still in the early testing phase. Items can be added/removed without notice and prices can increase/decrease.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
Bounty for https://github.com/BeeStation/BeeStation-Hornet/issues/627

File: code/modules/client/loadout/loadout_accessories.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/datum/gear/accessory
+	subtype_path = /datum/gear/accessory
+	slot = SLOT_NECK
+	sort_category = "Accessories"
+
+/datum/gear/accessory/scarf/red
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_accessories.dm:6]: 1000

File: code/modules/client/loadout/loadout_cosmestics.dm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/datum/gear/lipstick
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_cosmestics.dm:1]: 1200

File: code/modules/client/loadout/loadout_general.dm [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/datum/gear/dice
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_general.dm:1]: 300

File: code/modules/client/loadout/loadout_hat.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/datum/gear/hat
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_hat.dm:1]: 800

File: code/modules/client/loadout/loadout_hat.dm [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	subtype_path = /datum/gear/hat
+	slot = SLOT_HEAD
+	sort_category = "Headwear"
+	species_blacklist = list("plasmaman") //Their helmet takes up the head slot
+
+/datum/gear/hat/hhat_yellow
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_hat.dm:8]: 900

File: code/modules/client/loadout/loadout_hat.dm [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	display_name = "hardhat, orange"
+	path = /obj/item/clothing/head/hardhat/orange
+	allowed_roles = list("Chief Engineer", "Engineer")
+
+/datum/gear/hat/hhat_blue
+	display_name = "hardhat, blue"
+	path = /obj/item/clothing/head/hardhat/dblue
+	allowed_roles = list("Chief Engineer", "Engineer")
+
+/datum/gear/hat/that
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_hat.dm:24]: 1000

File: code/modules/client/loadout/loadout_hat.dm [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	display_name = "top hat"
+	path = /obj/item/clothing/head/that
+
+/datum/gear/hat/red_beret
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_hat.dm:29]: 2000

File: code/modules/client/loadout/loadout_ooc.dm [hunk 8]
```diff
@@ -0,0 +1,8 @@
+/datum/gear/ooc/char_slot
+	display_name = "extra character slot"
+	sort_category = "OOC"
+	description = "An extra charslot. Pretty self-explanatory."
+	cost = 10
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_ooc.dm]: 10000

File: code/modules/client/loadout/loadout_suit.dm [hunk 9]
```diff
@@ -0,0 +1,85 @@
+/datum/gear/suit
+	subtype_path = /datum/gear/suit
+	slot = SLOT_WEAR_SUIT
+	sort_category = "External Wear"
+
+//WINTER COATS
+/datum/gear/suit/coat
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_suit.dm:7]: 2000

File: code/modules/client/loadout/loadout_suit.dm [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+	path = /obj/item/clothing/suit/hooded/wintercoat/cargo
+	allowed_roles = list("Quartermaster", "Cargo Technician")
+
+/datum/gear/suit/coat/job/miner
+	display_name = "winter coat, miner"
+	path = /obj/item/clothing/suit/hooded/wintercoat/miner
+	allowed_roles = list("Miner")
+
+//JACKETS
+/datum/gear/suit/leather_jacket
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_suit.dm]: 2500

File: code/modules/client/loadout/loadout_uniform.dm [hunk 11]
```diff
@@ -0,0 +1,16 @@
+// Uniform slot
+/datum/gear/uniform
+	subtype_path = /datum/gear/uniform
+	slot = SLOT_W_UNIFORM
+	sort_category = "Uniforms and Casual Dress"
+
+/datum/gear/uniform/skirt
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_uniform.dm:7]: 1600

**Comment by Bastian0930** [comment]
Now I gotta find a way to hackerman beecoins for character slots that I don't use

**Comment by Crossedfall** [comment]
Test merged onto Golden.

**Pull request merged**